### PR TITLE
Enable back button if going from dapp to home & redirection problems

### LINKED ISSUE #1408: Enable back button if going from dapp to home
If I'm on a dapp and I tap the home button, I am unable to go back to the dapp I was previously on. Let's enable the back button if the user was previously on a dapp.

### PR DESCRIPTION
**Pull request by andrepimenta**
**Description**

This enables going back to the last url after you pressed home as mentioned in #1408. Also fixes a redirection problem, that happened sometimes when going to a dapp from home as mentioned on: [#1408 comment](https://github.com/MetaMask/metamask-mobile/issues/1408#issuecomment-607923956).

(Note: While trying to fix this, there were other small issues found with the navigation that are not addressed on the issue #1408 so we should test the browser navigation a bit more and maybe create another issue for refactoring it.)

**Issue**

Resolves #1408

### CODE REVIEW

File: app/components/Views/BrowserTab/index.js [hunk 1]
```diff
@@ -1069,8 +1069,17 @@ export class BrowserTab extends PureComponent {
...
+
 		const { current } = this.webview;
-		current && current.goBack();
+		const { lastUrlBeforeHome } = this.state;
+
+
+		if(this.isHomepage() && lastUrlBeforeHome){
+			this.go(lastUrlBeforeHome);
+		}else{
+			current && current.goBack();
```
Comment (danjm) [inline app/components/Views/BrowserTab/index.js:1080]: If `this.isHomepage()` is true and `lastUrlBeforeHome` is false, is it possible that `current.goBack()` could be called when there is nowhere to go back to? I am thinking of when `this.handleAndroidBackPress` is called.

**Review by danjm — CHANGES_REQUESTED** [review]
Left one question about a case we might want to handle. But overall, these change look correct to me.

**Comment by ibrahimtaveras00** [comment]
Positive test cases (happy flow) looks good 

Just a few issues I saw when comparing to what's on playstore and testflight (version 0.2.14 389)

**Issue 1:**
_These issues are only happening on Android, iOS works fine_

When on homepage, then navigating to a dapp via "Featured Projects", searching via search engine (duck duck go or google), or navigating directly to a URL, then pressing back button, the forward button does not appear as an option.

FYI Screen casts have the emulator with this branch on left, and real device with playstore on the right...

Via Featured Projects seen here = https://recordit.co/yZF2jPKcit

Steps to reproduce:
- on an android device go to browser and start on the homepage
- tap on a dapp from Featured Projects 
- once on the dapp page, tap on the back arrow 
- once back on the homepage, notice the forward arrow is not enabled 

Via search engine seen here = https://recordit.co/2JtrEla5yL

Steps to reproduce:
- on an android device go to browser and start on the homepage
- in the search text input field search anything or go directly to a URL 
- now tap the back arrow
- once back on the homepage, notice the forward arrow is not enabled 

**Issue 2:**
_This one happens on both Android and iOS_

When first arriving on homepage, or opening a new tab, and tapping the home button, the back arrow is enabled, and then tapping it shows the forward arrow, which keeps you on the homepage when tapping either.

Seen here = https://recordit.co/Flfx2NThCf

steps to reproduce:
- navigate to browser or just open a new tab
- tap the home button 
- notice the back arrow is enabled, now tap the back arrow and the forward arrow will be enabled

**Review by danjm — APPROVED** [review]

**Review by ibrahimtaveras00 — APPROVED** [review]
Fixes look good on both OS's, QA Passed 👍

**Pull request merged**